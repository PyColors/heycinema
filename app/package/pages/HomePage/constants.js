export const CHANGE_TITLE = 'boilerplate/Home/CHANGE_TITLE';
