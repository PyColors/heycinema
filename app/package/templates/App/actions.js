import {
  LOAD_MOVIES,
  LOAD_MOVIES_SUCCESS,
  LOAD_MOVIES_ERROR,
} from './constants';

/**
 * Load the movies, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_MOVIES
 */
export function loadMovies() {
  return {
    type: LOAD_MOVIES,
  };
}

/**
 * Dispatched when the movies are loaded by the request saga
 *
 * @param  {object} movies The movies data
 * @param  {string} title The current title
 *
 * @return {object}      An action object with a type of LOAD_MOVIES_SUCCESS passing the movies
 */
export function moviesLoaded(movies, title) {
  return {
    type: LOAD_MOVIES_SUCCESS,
    movies,
    title,
  };
}

/**
 * Dispatched when loading the movies fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of LOAD_MOVIES_ERROR passing the error
 */
export function repoLoadingError(error) {
  return {
    type: LOAD_MOVIES_ERROR,
    error,
  };
}
