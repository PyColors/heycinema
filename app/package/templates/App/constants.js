export const LOAD_MOVIES = 'heycinema/App/LOAD_MOVIES';
export const LOAD_MOVIES_SUCCESS = 'heycinema/App/LOAD_MOVIES_SUCCESS';
export const LOAD_MOVIES_ERROR = 'heycinema/App/LOAD_MOVIES_ERROR';
export const DEFAULT_LOCALE = 'en';
