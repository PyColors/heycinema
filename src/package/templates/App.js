import React from 'react';
import SearchPage from '../pages/SearchPage';

const App = () => (

  <main className='content'>
  <header className="mui-appbar mui--z1 content__appbar">
  <div className="mui-container">
    <div className="mui--appbar-height">
      <div className="mui--text-title content__title">
        <span className="content__title-1">hey</span>
        <span className="content__title-2">cinema</span>
      </div>
    </div>
  </div>
  </header>

  <div id="content-wrapper" className="mui--text-center">
    <div className="mui--appbar-height"></div>
    <SearchPage />
  </div>

  <footer>
    <div className="mui-container mui--text-center">
      Made with â™¥ 
    </div>
  </footer>

  </main>
)

export default App;

