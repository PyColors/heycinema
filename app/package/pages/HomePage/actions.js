import { CHANGE_TITLE } from './constants';

/**
 * Changes the input field of the form
 *
 * @param  {name} name The new text of the input field
 *
 * @return {object}    An action object with a type of CHANGE_TITLE
 */
export function changeTitle(name) {
  return {
    type: CHANGE_TITLE,
    name
  };
}
