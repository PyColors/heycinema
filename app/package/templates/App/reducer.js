import { LOAD_MOVIES_SUCCESS, LOAD_MOVIES, LOAD_MOVIES_ERROR } from './constants';

// The initial state of the App
export const initialState = {
  loading: false,
  error: false,
  currentUser: false,
  userData: {
    movies: false,
  },
};

function appReducer(state = initialState, action) {
  switch (action.type) {
    case LOAD_MOVIES: {
      const newState = {
        ...state,
        loading: true,
        error: false,
        userData: {
          movies: false,
        },
      };

      return newState;
    }
    case LOAD_MOVIES_SUCCESS: {
      const newState = {
        ...state,
        loading: false,
        userData: {
          movies: action.movies,
        },
        currentUser: action.title,
      };
      return newState;
    }

    case LOAD_MOVIES_ERROR: {
      return { ...state, error: action.error, loading: false };
    }
    default:
      return state;
  }
}

export default appReducer;
