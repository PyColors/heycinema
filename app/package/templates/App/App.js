import React from 'react';
import { Helmet } from 'react-helmet';
import { Switch, Route } from 'react-router-dom';
import Header from '../../components/organisms/Header';
import HomePage from '../../pages/HomePage/Loadable';
import AboutPage from '../../pages/AboutPage/Loadable';
import NotFoundPage from '../../pages/NotFoundPage/Loadable';
import Footer from '../../components/organisms/Footer';

const App = () => (
  <main className="content">
    <Header />
    <Helmet
      titleTemplate="%s - Movies"
      defaultTitle="Movies"
    >
      <meta name="description" content="Movies" />
    </Helmet>
    <Switch>
      <Route exact path="/" component={HomePage} />
      <Route exact path="/about" component={AboutPage} />
      <Route path="" component={NotFoundPage} />
    </Switch>
    <Footer />
  </main>
);

export default App;
