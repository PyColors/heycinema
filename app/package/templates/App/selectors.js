import { createSelector } from 'reselect';
import { initialState } from './reducer';

const selectGlobal = (state) => state.global || initialState;

const selectRoute = (state) => state.router;

const makeSelectLoading = () => createSelector(
  selectGlobal,
  (globalState) => globalState.loading
);

const makeSelectError = () => createSelector(
  selectGlobal,
  (globalState) => globalState.error
);

const makeSelectMovies = () => createSelector(
  selectGlobal,
  (globalState) => globalState.userData.movies
);

const makeSelectLocation = () => createSelector(
  selectRoute,
  (routeState) => routeState.location
);

export {
  selectGlobal,
  makeSelectLoading,
  makeSelectError,
  makeSelectMovies,
  makeSelectLocation,
};
