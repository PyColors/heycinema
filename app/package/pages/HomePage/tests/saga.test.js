// Tests for HomePage sagas
import { put, takeLatest } from 'redux-saga/effects';

import { LOAD_MOVIES } from '../../../templates/App/constants';
import { moviesLoaded, moviesLoadingError } from '../../../templates/App/actions';

// import omdDData, saga manages watcher lifecycle
import omdDData, { getMovies } from '../saga';

const title = 'Titanic';

/* eslint-disable redux-saga/yield-effects */
describe('getMovies Saga', () => {
  let getMoviesGenerator;

  // Test twice, once for a successful load and once for an unsuccessful one
  // All the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    getMoviesGenerator = getMovies();

    const selectDescriptor = getMoviesGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = getMoviesGenerator.next(title).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the moviesLoaded action if it requests the data successfully', () => {
    const response = [{
      name: 'First movie',
    }, {
      name: 'Second movie',
    }];
    const putDescriptor = getMoviesGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(moviesLoaded(response, title)));
  });

  it('should call the moviesLoadingError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getMoviesGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(moviesLoadingError(response)));
  });
});

describe('omdDDataSaga Saga', () => {
  const omdDDataSaga = omdDData();

  it('should start task to watch for LOAD_MOVIES action', () => {
    const takeLatestDescriptor = omdDDataSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(LOAD_MOVIES, getMovies));
  });
});
