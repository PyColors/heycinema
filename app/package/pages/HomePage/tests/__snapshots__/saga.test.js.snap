// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMovies Saga should call the moviesLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getMovies Saga should call the moviesLoadingError action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "http://www.omdbapi.com/?t=Titanic&apikey=ebcafcfe",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getMovies Saga should dispatch the moviesLoaded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getMovies Saga should dispatch the moviesLoaded action if it requests the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "http://www.omdbapi.com/?t=Titanic&apikey=ebcafcfe",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
